### PR TITLE
Add fmt (cppformat) 3.0.0

### PR DESCRIPTION
**Pull request by dpantele**
fmt (previously cppformat) is a nice, safe and fast printf/iostreams replacement

Url: http://fmtlib.net/

It defines two exported targets: fmt and fmt-header-only. It is slightly different from what is used in hunter usually (something like fmt::fmt and fmt::header-only), but otherwise everything works out-of-box.

### CODE REVIEW

**Comment by ruslo** [comment]
Thanks, testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.429
* https://travis-ci.org/ingenue/hunter/builds/130544210

File: cmake/projects/fmt/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+hunter_cmake_args(
+    fmt
+    CMAKE_ARGS
+        FMT_DOC=OFF
+        FMT_TEST=OFF
+)
+
+# Pick a download scheme
+hunter_pick_scheme(DEFAULT url_sha1_cmake) # use scheme for cmake projects
+
```
Comment (ruslo) [inline cmake/projects/fmt/hunter.cmake]: You can make it [cacheable](https://github.com/ruslo/hunter/wiki/dev.modules#hunter_cacheable) so we can upload this library.

**Comment by ruslo** [comment]
Seems like Android failed:
* https://travis-ci.org/ingenue/hunter/jobs/130544214

@dpantele You can:
* fix Android and send new version of `fmt`
* send pull request to [pkg.fmt](https://github.com/ingenue/hunter/tree/pkg.fmt) branch to `ingenue/hunter` with excluding Android toolchain from build. Leave a comments with link to the broken Travis CI builds so others can find easily what is wrong ([example](https://github.com/ingenue/hunter/blob/4c4188b3d580f0e11cbc3e4e1a54c11616e870ae/.travis.yml#L102))

**Comment by ruslo** [comment]
Same for `vs-9-2008` on Windows:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.429

**Comment by dpantele** [comment]
Pushed some fixes.

Android compilation fix is to be merged: https://github.com/fmtlib/fmt/pull/328 , but I am not sure when it will get released.

**Comment by ruslo** [comment]
> Android compilation fix is to be merged: fmtlib/fmt#328 , but I am not sure when it will get released

* We can wait for release
* You can release your own forked version
* You can exclude Android for now until release

@dpantele up to you

**Comment by dpantele** [comment]
Ok, I don't need android, so just removed the broken build. Will make another pr when that get fixed.

**Comment by ruslo** [comment]
New testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.439
* https://travis-ci.org/ingenue/hunter/builds/130898790

**Comment by ruslo** [comment]
> Ok, I don't need android, so just removed the broken build: ingenue#16. Will make another pr when that get fixed. 

Need to do the same with `vs-9-2008` I guess:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.429

**Comment by dpantele** [comment]
Sorry, I probably should not have amended this pull-request. This is the right commit to test: https://github.com/ruslo/hunter/pull/413/commits/4264b055e4da3fb37d454c564f271a02130153b7

**Comment by dpantele** [comment]
I have removed auto from the test there.

**Comment by ruslo** [comment]
> This is the right commit to test: 4264b05

@dpantele Not sure I follow. This is the exact commit I was tested. It works ok, the only issue left is `vs-9-2008` build. You can try to fix it or exclude similar to https://github.com/ingenue/hunter/pull/16.

**Comment by dpantele** [comment]
That is strange: https://ci.appveyor.com/project/ingenue/hunter/build/1.0.429/job/s92y4md1ivgtba8m refers https://github.com/ingenue/hunter/commit/71da86bf1d0ec07e74328dc847fa59e610637e72 , which is different from https://github.com/ruslo/hunter/commit/4264b055e4da3fb37d454c564f271a02130153b7 . VS-2008 is complaining about the `auto`, which was removed from `fmtexample.cpp`

**Comment by dpantele** [comment]
Ah, actually this is the result of the last test: https://ci.appveyor.com/project/ingenue/hunter/build/1.0.439

Looks like it passes.

**Comment by ruslo** [comment]
> Looks like it passes

Yep, looks good. Haven't figured out it does relates :)

Thanks, merged.

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.451
* https://travis-ci.org/ingenue/hunter/builds/131272829

**Comment by ruslo** [comment]
> Uploading

Done. Released: https://github.com/ruslo/hunter/releases/tag/v0.14.22

@dpantele feel free to add wiki page for `fmt`. See [example](https://github.com/ruslo/hunter/wiki/pkg.zeromq).

**Comment by dpantele** [comment]
Ok, thanks, [added the page](https://github.com/ruslo/hunter/wiki/pkg.fmt).

**Pull request merged**